rasa_core/server.py

```diff
@@ -508,8 +507,24 @@ def evaluate_stories():
                          "Evaluation could not be created. Error: {}"
                          "".format(e))
 
-    @app.route("/domain",
-               methods=['GET', 'OPTIONS'])
+    @app.route("/jobs", methods=['POST', 'OPTIONS'])
+    @requires_auth(app, auth_token)
+    @cross_origin(origins=cors_origins)
+    def train_stack():
+        from rasa.cli.train import train
+        from argparse import Namespace
+
+        """Train a Rasa Stack model."""
+        args = Namespace(**request.get_json())
+        print("got args", args, request.get_json())
+        try:
+            model_name = train(args)
+            return jsonify({'info': 'new model trained', 'model': model_name})
```

Comment:
I don't think this is how this should work: we need this function when running on the server as well to train models in a worker. In that case the training data and the domain file won't be on the worker server, but rather should be transmitted with the request, as we do it in Rasa NLU (where we post the config and the training data to the endpoint - not only file names).

In addition to that, I think this should return the model instead of json